File: mobile/ios/Atacte/CalendarWidgetModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CalendarWidgetModule, NSObject)

RCT_EXTERN_METHOD(saveAuthToken:(NSString *)token apiUrl:(NSString *)apiUrl resolver:(RCTPromiseResolveBlock)resolver rejecter:(RCTPromiseRejectBlock)rejecter)
RCT_EXTERN_METHOD(clearAuthToken:(RCTPromiseResolveBlock)resolver rejecter:(RCTPromiseRejectBlock)rejecter)

@end

File: mobile/ios/Atacte/CalendarWidgetModule.swift
import Foundation
import React

@objc(CalendarWidgetModule)
class CalendarWidgetModule: NSObject {
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc
  func saveAuthToken(_ token: String, apiUrl: String, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
    let sharedDefaults = UserDefaults(suiteName: "group.com.sentro.mobile")
    sharedDefaults?.set(token, forKey: "auth_token")
    sharedDefaults?.set(apiUrl, forKey: "api_url")
    sharedDefaults?.synchronize()
    resolver(true)
  }
  
  @objc
  func clearAuthToken(_ resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
    let sharedDefaults = UserDefaults(suiteName: "group.com.sentro.mobile")
    sharedDefaults?.removeObject(forKey: "auth_token")
    sharedDefaults?.removeObject(forKey: "api_url")
    sharedDefaults?.synchronize()
    resolver(true)
  }
}
